packages/flutter/lib/src/cupertino/switch.dart

```diff
@@ -352,283 +565,445 @@ class _CupertinoSwitchState extends State<CupertinoSwitch> with TickerProviderSt
+    if (_needsPositionAnimation) {
+      _needsPositionAnimation = false;
+      animateToValue();
+    }
+
     final CupertinoThemeData theme = CupertinoTheme.of(context);
+
     final Color activeColor = CupertinoDynamicColor.resolve(
       widget.activeColor
       ?? ((widget.applyTheme ?? theme.applyThemeToAll) ? theme.primaryColor : null)
       ?? CupertinoColors.systemGreen,
       context,
     );
+
     final (Color onLabelColor, Color offLabelColor)? onOffLabelColors =
-        MediaQuery.onOffSwitchLabelsOf(context)
-            ? (
-                CupertinoDynamicColor.resolve(
-                  widget.onLabelColor ?? CupertinoColors.white,
-                  context,
-                ),
-                CupertinoDynamicColor.resolve(
-                  widget.offLabelColor ?? _kOffLabelColor,
-                  context,
-                ),
-              )
-            : null;
-    if (needsPositionAnimation) {
-      _resumePositionAnimation();
-    }
-    return MouseRegion(
-      cursor: isInteractive && kIsWeb ? SystemMouseCursors.click : MouseCursor.defer,
-      child: Opacity(
-        opacity: widget.onChanged == null ? _kCupertinoSwitchDisabledOpacity : 1.0,
-        child: FocusableActionDetector(
-          onShowFocusHighlight: _onShowFocusHighlight,
-          actions: _actionMap,
-          enabled: isInteractive,
-          focusNode: widget.focusNode,
-          onFocusChange: widget.onFocusChange,
-          autofocus: widget.autofocus,
-          child: _CupertinoSwitchRenderObjectWidget(
-            value: widget.value,
-            activeColor: activeColor,
-            trackColor: CupertinoDynamicColor.resolve(widget.trackColor ?? CupertinoColors.secondarySystemFill, context),
-            thumbColor: CupertinoDynamicColor.resolve(widget.thumbColor ?? CupertinoColors.white, context),
-            // Opacity, lightness, and saturation values were approximated with
-            // color pickers on the switches in the macOS settings.
-            focusColor: CupertinoDynamicColor.resolve(
-              widget.focusColor ??
-              HSLColor
-                    .fromColor(activeColor.withOpacity(0.80))
-                    .withLightness(0.69).withSaturation(0.835)
+      MediaQuery.onOffSwitchLabelsOf(context)
+        ? (CupertinoDynamicColor.resolve(
+            widget.onLabelColor ?? CupertinoColors.white,
+            context,
+          ),
+          CupertinoDynamicColor.resolve(
+            widget.offLabelColor ?? _kOffLabelColor,
+            context,
+          ),
+        )
+        : null;
+
+    // Colors need to be resolved in selected and non selected states separately
+    // so that they can be lerped between.
+    final Set<WidgetState> activeStates = states..add(WidgetState.selected);
+    final Set<WidgetState> inactiveStates = states..remove(WidgetState.selected);
+
+    final Color effectiveActiveThumbColor =
+      _resolveThumbColor(widget.thumbColor, activeStates)
```

Comment:
nit: I think for assignments like this and below, we don't need to break the line even though it is a little longer than 80 characters.
https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#prefer-a-maximum-line-length-of-80-characters